new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java

```diff
@@ -0,0 +1,555 @@
+                .replicas(replicaCount)
+                .serverStartState("RUNNING"))
+            .configuration(new Configuration()
+                .model(new Model()
+                    .domainType("WLS")
+                    .runtimeEncryptionSecret(encryptionSecretName))));
+
+    logger.info("Create domain custom resource for domainUID {0} in namespace {1}",
+        domainUID, domainNamespace);
+    createDomain(domain);
+
+    // wait for the domain to exist
+    logger.info("Check for domain custom resouce in namespace {0}", domainNamespace);
+    waitForCondition(domainExists(domainUID, DOMAIN_VERSION, domainNamespace));
+
+
+    // check admin server pod exist
+    logger.info("Check for admin server pod {0} existence in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodCreated(adminServerPodName);
+
+    // check managed server pods exists
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Check for managed server pod {0} existence in namespace {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkPodCreated(managedServerPrefix + i);
+    }
+
+    // check admin server pod is running
+    logger.info("Wait for admin server pod {0} to be ready in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodRunning(adminServerPodName);
+
+    // check managed server pods are running
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed server pod {0} to be ready in namespace {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkPodRunning(managedServerPrefix + i);
+    }
+
+    logger.info("Check admin service {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkServiceCreated(adminServerPodName);
+
+    // check managed server services created
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Check managed server service {0} is created in namespace {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkServiceCreated(managedServerPrefix + i);
+    }
+
+  }
+
+  @AfterEach
+  public void tearDown() {
+
+    // Delete domain custom resource
+    logger.info("Delete domain custom resource in namespace {0}", domainNamespace);
+    assertThatCode(
+        () -> deleteDomainCustomResource(domainUID, domainNamespace))
+        .as("Test that deleteDomainCustomResource doesn not throw an exception")
+        .withFailMessage("delete domain custom resource failed")
+        .doesNotThrowAnyException();
+
+    logger.info("Deleted Domain Custom Resource " + domainUID + " from " + domainNamespace);
+
+    // delete the domain image created for the test
+    if (miiImage != null) {
+      deleteImage(miiImage);
+    }
+
+  }
+
+  /**
+   * Uninstall Operator, delete service account, domain namespace and
+   * operator namespace.
+   */
+  @AfterAll
+  public void tearDownAll() {
+    // uninstall operator release
```

Comment:
this should be handled by generic code in an extension, that will clean up the namespace(s) used by any test class that is finished

Reply:
Sankar or someone is working on clean up common class. This can be changed when that is ready, till that time don't want to leave the domain, operator, namespaces running after the test completes and hence here.